mapfishapp/src/main/webapp/app/js/GEOR_Querier.js

```diff
@@ -0,0 +1,330 @@
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with geOrchestra.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/*
+ * Notes about the -AT-requires and -AT-include annotations below:
+ *
+ * - -AT-requires OpenLayers/Filter/Comparison.js is required because
+ *   Styler/widgets/form/ComparisonComboBox.js, which is
+ *   included by Styler/widgets/FilterPanel.js, requires it
+ * - -AT-requires OpenLayers/Filter/Spatial.js is required because
+ *   Styler/widgets/form/SpatialComboBox.js, which is
+ *   included by Styler/widgets/SpatialFilterPanel.js, requires it
+ * - -AT-include OpenLayers/Filter/Logical.js is required because
+ *   Styler/widgets/FilterBuilder.js uses OpenLayers.Filter.Logical
+ */
+
+/*
+ * @requires OpenLayers/Filter/Comparison.js
+ * @requires OpenLayers/Filter/Spatial.js
+ * @include OpenLayers/Filter/Logical.js
+ * @include OpenLayers/Format/Filter.js
+ * @include OpenLayers/Format/XML.js
+ // see https://github.com/georchestra/georchestra/issues/482 :
+ * @include OpenLayers/Format/WKT.js
+ * @include OpenLayers/Control/ModifyFeature.js
+ * @include OpenLayers/Control/DrawFeature.js
+ * @include OpenLayers/Handler/Point.js
+ * @include OpenLayers/Handler/Path.js
+ * @include OpenLayers/Handler/Polygon.js
+ * @include GeoExt/data/LayerRecord.js
+ * @include GeoExt/data/AttributeStore.js
+ * @include Styler/widgets/FilterBuilder.js
+ * @include GEOR_waiter.js
+ * @include GEOR_ows.js
+ * @include GEOR_util.js
+ */
+
+/* globals Ext, GeoExt, OpenLayers, GEOR */
+
+Ext.namespace("GEOR");
+
+
+GEOR.Querier = Ext.extend(Ext.Window, {
+
+    layout: "fit",
+    closeAction: "close",
+    border: false,
+ 
+    /**
+     * Property: filterbuilder
+     * {String} The local Styler.FilterBuilder instance
+     */
+    filterbuilder: null,
+
+    /**
+     * Property: pseudoRecord
+     * {Object} Object with typeName & owsURL properties
+     */
+    pseudoRecord: null,
+
+    /**
+     * Property: geometryName
+     * {String} The geometry column name
+     */
+    geometryName: null,
+
+    /**
+     * Property: record
+     * {Ext.data.Record} The WMS or WFS layer record
+     */
+    record: null,
+
+    /**
+     * Property: map
+     * {OpenLayers.Map} The map instance.
+     */
+    map: null,
+
+    /**
+     * Property: layer
+     * {OpenLayers.Layer.Vector} The layer on which geometries can be drawn
+     */
+    layer: null,
+
+    /**
+     * Property: attributeStore
+     * {GeoExt.data.AttributeStore} 
+     */
+    attributeStore: null,
+
+    /**
+     * Property: layerFields
+     * {Array} an array of fields for the current layer
+     * It is extracted from the WFS DescribeFeatureType operation
+     */  
+    layerFields: null,
+
+    /*
+     * Method: initComponent.
+     * Overridden constructor. Set up widgets and lay them out
+     */
+    initComponent: function() {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Function 'initComponent' has a complexity of 5.](https://www.codacy.com/app/georchestra/georchestra/file/2338282099/issues/source?bid=3200181&fileBranchId=3209168#l113)